framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -1003,6 +1003,9 @@ protected boolean visitWildcardSupertype(
 
     protected boolean visitWildcardSubtype(
             AnnotatedWildcardType subtype, AnnotatedTypeMirror supertype, VisitHistory visited) {
+        if (subtype.isUninferredTypeArgument()) {
+            return ignoreUninferredTypeArguments;
```

Comment:
Missing errors:
I added checking uninferred wildcards against declared types without type arguments and now the tests are issuing the expected warnings.

Issuing warning/error if type argument inference fails:
I've opened another pull request for this #1343. We should move the discussion there.  (And we can talk about this at today's meeting.)